lib/mosaic-routing/src/main/java/org/eclipse/mosaic/lib/routing/pt/PtRouting.java

```diff
@@ -0,0 +1,183 @@
+
+        final Request ghRequest = new Request(
+                request.getStartingGeoPoint().getLatitude(),
+                request.getStartingGeoPoint().getLongitude(),
+                request.getTargetGeoPoint().getLatitude(),
+                request.getTargetGeoPoint().getLongitude()
+        );
+//        ghRequest.setBlockedRouteTypes(request.getRoutingParameters().excludedPtModes);//FIXME generalize this
+        ghRequest.setEarliestDepartureTime(departureTime);
+        ghRequest.setWalkSpeedKmH(request.getRoutingParameters().getWalkingSpeedMps() * 3.6);
+
+        final Future<GHResponse> responseFuture = routingExecution.submit(() -> ptRouter.route(ghRequest));
+        final GHResponse route;
+        try {
+            final StopWatch sw = new StopWatch();
+            sw.start();
+            route = responseFuture.get(30, TimeUnit.SECONDS);
+            sw.stop();
+            LOG.debug("Took {} ms to calculate public transport route.", sw.getMillis());
+        } catch (TimeoutException e) {
+            responseFuture.cancel(true);
+            throw new RuntimeException("Could not finish route calculation. Exceeded timeout.");
+        } catch (InterruptedException | ExecutionException e) {
+            throw new RuntimeException("Could not finish route calculation. Exceeded timeout.", e);
+        }
+
+        final List<MultiModalLeg> legs = convertToMultiModalLegs(route.getBest());
+
+        return new PtRoutingResponse(new MultiModalRoute(legs));
+    }
+
+    private List<MultiModalLeg> convertToMultiModalLegs(ResponsePath ghBestRoute) {
+        List<MultiModalLeg> legs = new ArrayList<>();
+        for (Trip.Leg leg : ghBestRoute.getLegs()) {
+            if (leg instanceof Trip.PtLeg ptLeg) {
+                List<PtLeg.PtStop> newStops = new ArrayList<>();
+                for (Trip.Stop stop : ptLeg.stops) {
+                    newStops.add(new PtLeg.PtStop(GeoPoint.lonLat(stop.geometry.getX(), stop.geometry.getY()), fromScheduleTime(stop.departureTime), fromScheduleTime(stop.arrivalTime)));
+                }
+                legs.add(new MultiModalLeg(
+                        new PtLeg(newStops), newStops.get(0).departureTime(), Iterables.getLast(newStops).arrivalTime())
+                );
+            } else if (leg instanceof Trip.WalkLeg walkLeg) {
+                List<GeoPoint> waypoints = new ArrayList<>();
+                for (Coordinate coordinate : walkLeg.geometry.getCoordinates()) {
+                    waypoints.add(GeoPoint.lonLat(coordinate.x, coordinate.y));
+                }
+                legs.add(new MultiModalLeg(new WalkLeg(waypoints), fromScheduleTime(leg.getDepartureTime()), fromScheduleTime(leg.getArrivalTime())));
+            }
+        }
+        return legs;
+    }
+
+    private Instant toScheduleTime(long simTime) {
+        return scheduleDateTime.plusNanos(simTime).atZone(timeZone).toInstant();
+    }
+
+    private Long fromScheduleTime(Date date) {
```

Comment:
```suggestion
    private Long fromScheduleTime(@Nonnull Date date) {
```

Reply:
Erm, you mean `@Nullable`? The method is able to handle the `null` case.